pkg/util/metric.go

```diff
@@ -0,0 +1,62 @@
+package util
+
+import (
+	"time"
+	"encoding/json"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	metricDuration = 5
+)
+
+// Rebalance Metric Config
+type RebalanceMetricConfig struct {
+	TopicName          string  `json:"topicName"`
+	ClusterName        string  `json:"clusterName"`
+	ClusterEnvironment string  `json:"clusterEnvironment"`
+	ToRemove           []int   `json:"toRemove"`
+	RebalanceStatus    string  `json:"rebalanceStatus"`
+}
+
+// Apply Metric Config
+type ApplyMetricConfig struct {
+	CurrRound          int     `json:"round"`
+	TotalRounds        int     `json:"totalRounds"`
+	TopicName          string  `json:"topicName"`
+	ClusterName        string  `json:"clusterName"`
+	ClusterEnvironment string  `json:"clusterEnvironment"`
+	ToRemove           []int   `json:"toRemove"`
+}
+
+// Print Metrics for each iteration
+// This can be configured but that will need lots of TopicConfig struct changes
+// For now, emitting metrics every 5seconds. refer metricDuration
```

Comment:
5 seconds seems excessively frequent, especially considering sleep-loop-duration defaults to 10s. I think 30s or even 60s should be sufficient to monitor processes expected to take hours.

Also, rather than emitting metrics on some timer, what do you think of the idea of emitting them when transitions occur?